src/screens/DiscoveryScreenV2/elements/Search.js

```diff
@@ -289,17 +280,30 @@ const styles = StyleSheet.create({
     marginRight: 11,
     ...FONTS.h3
   },
-  animatedViewContainer: {
+  animatedViewContainer: (hideBackIcon) => ({
     flexDirection: 'row',
     backgroundColor: 'white',
     marginTop: 0,
     zIndex: 10,
     height: dimen.size.DISCOVERY_HEADER_HEIGHT,
     paddingTop: 7,
     paddingBottom: 7,
-    borderBottomWidth: 1,
+    borderBottomWidth: hideBackIcon ? 0 : 1,
     borderBottomColor: COLORS.alto
-  }
+  })
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The style `animatedViewContainer` now depends on the `hideBackIcon` prop. This could potentially change the appearance of the component. Ensure that this change is reflected correctly in the UI.
```diff
- borderBottomWidth: 1,
+ borderBottomWidth: hideBackIcon ? 0 : 1,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->